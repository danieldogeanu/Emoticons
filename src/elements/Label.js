import React from 'react';

function Label(props) {
	return (
		<span className={'Label ' + props.label.class}>
			{props.label.name}
		</span>
	);
}

export default Label;
