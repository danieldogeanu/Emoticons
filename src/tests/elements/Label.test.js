import React from 'react';
import {render} from '@testing-library/react';
import Label from '../../elements/Label';

describe('Label Element', () => {

	it('renders list column label properly', () => {
		const labelObj = {class: 'test', name: 'Test'};
		const {getByText} = render(<Label label={labelObj} />);
		const renderedLabel = getByText(labelObj.name);

		expect(renderedLabel).toBeInTheDocument();
		expect(renderedLabel).toHaveTextContent(labelObj.name);
		expect(renderedLabel).toHaveClass('Label', labelObj.class);
		expect(renderedLabel).toMatchSnapshot();
	});

});
