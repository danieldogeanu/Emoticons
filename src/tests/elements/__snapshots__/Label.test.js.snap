// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Element renders list column label properly 1`] = `
<span
  class="Label test"
>
  Test
</span>
`;
